File: calcE.py
import sys


def ecalc():
    n = int(sys.argv[1])
    m = n
    dic = {}

    sys.stdout.write("2.")

    for j in reversed(range(2, m)):
        dic[j] = 1

    for i in range(1, n):
        carry = 0
        for j in reversed(range(2, m)):
            temp = dic[j] * 10 + carry
            carry = temp / j
            dic[j] = temp - carry * j

        sys.stdout.write(str(carry))

ecalc()

File: calcE.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, const char *argv[])
{
    int n = atoi(argv[1]);
    int m = n;
    int a[m];

    int i, j, carry, temp;
    
    printf("%d.", 2);
    for (j = m; j >= 2; j--) {
        a[j] = 1;
    }

    for (i = 1; i <= n; i++) {
        carry = 0;
        for (j = m; j >= 2; j--) {
            temp = a[j] * 10 + carry; 
            carry = temp / j;
            a[j] = temp - carry * j;
        }
        printf("%d", carry);
    }
    
    return 0;
}
